src/pfe/file-watcher/server/test/functional-test/configs/project.config.ts

```diff
@@ -11,64 +11,99 @@
 /**
  * This file for different project specific (i.e liberty, spring, docker) capabaility configurations.
  */
+
+export const needManualReset: any = {
+    "docker": {
+        "runningBuildQueue": false,
+        "buildStatus": false
+    },
+    "nodejs": {
+        "runningBuildQueue": true,
+        "buildStatus": true
+    },
+};
+
 export const projectCapabilities: any = {
     "docker": {
         "startModes": ["run"],
         "controlCommands": []
+    },
+    "nodejs": {
+        "startModes": ["run", "debugNoInit"],
+        "controlCommands": ["restart"]
     }
 };
 
+export const startModes: Array<string> = ["run", "debug", "debugNoInit"];
+
 export const restartCapabilities: any = {
-    "docker": false
+    "docker": {
+        [startModes[0]]: false,
+        [startModes[1]]: false,
+        [startModes[2]]: false,
+    },
+    "nodejs": {
+        [startModes[0]]: true,
+        [startModes[1]]: false,
+        [startModes[2]]: true,
+    },
 };
 
 export const debugCapabilities: any = {
-    "docker": false
+    "docker": false,
+    "nodejs": true
 };
 
 export const mavenProfileCapabilities: any = {
     "docker":  false,
+    "nodejs": false
 };
 
-export const exposedDebugPorts: any = {
-    "docker": ["1000"]
-};
+export const exposedDebugPorts = ["1000"];
 
 export const defaultContextRoot: any = {
     "lagom": "/api/hello/you"
 };
 
 export const defaultHealthCheckEndPoint: any = {
     "go": "/",
-    "python": "/"
+    "lagom": "/",
+    "python": "/",
+    "nodejs": "/"
 };
 
 export const oneExposedPortOnly: any = {
     "go": {
         "local": true,
         "kube": true
     },
+    "lagom": {
+        "local": false,
+        "kube": true
+    },
     "python": {
         "local": true,
         "kube": true
     },
-    "lagom": {
+    "nodejs": {
         "local": false,
         "kube": true
     }
 };
 
 export const defaultInternalPorts: any = {
     "go": "8000",
+    "lagom": "9000",
     "python": "5000",
-    "lagom": "9000"
+    "nodejs": "3000"
```

Comment:
Can we use the `nodejsProject.getDefaultAppPort()` to get the defaiult port.. we don't have this for docker project types, but at least we're not hard coding it for non-Docker type projects. Pls consider making the change if you have an object for debug ports as well

Reply:
Nice. I like this feedback 👍